modules/test/playwright/tests/frontend-js-components-web/translationManager.spec.ts

```diff
@@ -0,0 +1,392 @@
+import {dataApiHelpersTest} from '../../fixtures/dataApiHelpersTest';
+import {featureFlagsTest} from '../../fixtures/featureFlagsTest';
+import {isolatedSiteTest} from '../../fixtures/isolatedSiteTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {clickAndExpectToBeHidden} from '../../utils/clickAndExpectToBeHidden';
+import {clickAndExpectToBeVisible} from '../../utils/clickAndExpectToBeVisible';
+import {jsComponentsSamplePageTest} from './fixtures/jsComponentsSamplePageTest';
+
+export const test = mergeTests(
+	dataApiHelpersTest,
+	apiHelpersTest,
+	featureFlagsTest({
+		'LPS-178052': {enabled: true},
+	}),
+	jsComponentsSamplePageTest,
+	isolatedSiteTest,
+	loginTest()
+);
+
+test.beforeEach(
+	'Setup site and JS Components Sample widget',
+	async ({apiHelpers, jsComponentsSamplePage, page, site}) => {
+		await test.step('Create a content site and the js components sample widget', async () => {
+			await jsComponentsSamplePage.setupJSComponentsSampleWidget({
+				apiHelpers,
+				site,
+			});
+		});
+
+		await test.step('Select Translation Manager tab', async () => {
+			await jsComponentsSamplePage.selectTab(
+				'Translation Manager',
+				page.getByRole('heading', {name: 'AUI Tag'})
+			);
+		});
+	}
+);
+
+test.describe('Translation Manager', () => {
+	test(
+		'Assert the localization dropdown is rendered alongside the localized input',
+		{tag: '@LPD-47235'},
+		async ({jsComponentsSamplePage}) => {
+			await test.step('Assert the localized input is rendered', async () => {
+				expect(
+					jsComponentsSamplePage.adminLocalizedInputContainer
+				).toBeVisible();
+			});
+
+			await test.step('Assert the localization dropdown is rendered', async () => {
+				const {
+					translationManagerCatalanChoice,
+					translationManagerTriggerButton,
+				} = jsComponentsSamplePage;
+
+				expect(translationManagerTriggerButton).toBeVisible();
+
+				await clickAndExpectToBeVisible({
+					target: translationManagerCatalanChoice,
+					trigger: translationManagerTriggerButton,
+				});
+			});
+		}
+	);
+
+	test(
+		'Assert the click on a translation updates the localized input',
+		{tag: '@LPD-47235'},
+		async ({jsComponentsSamplePage, page}) => {
+			const {
+				adminLocalizedInputContainer,
+				localizedInput,
+				translationManagerCatalanChoice,
+				translationManagerTriggerButton,
+			} = jsComponentsSamplePage;
+
+			await test.step('Fill the localized input with a translation', async () => {
+				await expect(adminLocalizedInputContainer).toBeVisible();
+
+				await localizedInput.fill('Translation');
+			});
+
+			await test.step('Click on a translation', async () => {
+				await clickAndExpectToBeVisible({
+					target: translationManagerCatalanChoice,
+					trigger: translationManagerTriggerButton,
+				});
+
+				await translationManagerCatalanChoice.click();
+
+				const translationText = page
+					.locator('text=Translation')
+					.first();
```

Comment:
This can accidentally find something else on the page. To make sure sample fill is unique, you can use `getRandomString()` util.